search/services/index/highlighting.py

```diff
@@ -0,0 +1,269 @@
+"""
+Provide hit highlighting to the search.
+
+Highlighting requires amendation of the query as well as post-processing of
```

Comment:
Good feature, looking good with wildcards.
https://beta.arxiv.org/search/advanced?advanced=1&terms-0-operator=AND&terms-0-term=elec*&terms-0-field=title&classification-eess=y&classification-physics_archives=all&date-filter_by=specific_year&date-year=2017&date-from_date=&date-to_date=&size=50&order=
https://beta.arxiv.org/search/advanced?advanced=1&terms-0-operator=AND&terms-0-term=elec*&terms-0-field=abstract&terms-1-operator=AND&terms-1-term=elec*&terms-1-field=comments&classification-physics=y&classification-physics_archives=all&date-filter_by=specific_year&date-year=2017&date-from_date=&date-to_date=&size=50&order=
Sometimes multiple wildcards confuse highlighting, or from tests below I gather this is expected behavior.
https://beta.arxiv.org/search/advanced?advanced=1&terms-0-operator=AND&terms-0-term=H*2*O&terms-0-field=all&classification-physics_archives=all&date-filter_by=all_dates&date-year=2017&date-from_date=2002&date-to_date=2003&size=50&order=

Reply:
This is indeed odd, since an explicit search on title does yield highlighting: https://beta.arxiv.org/search/advanced?advanced=&terms-0-operator=AND&terms-0-term=H*2*O&terms-0-field=title&terms-1-operator=AND&terms-1-term=phonons&terms-1-field=title&classification-physics_archives=all&date-filter_by=all_dates&date-year=2017&date-from_date=2002&date-to_date=2003&size=50&order=

Reply:
Opened https://github.com/cul-it/arxiv-search/issues/183 so that this doesn't get lost